controllers/component_dependency_update_controller.go

```diff
@@ -488,6 +539,145 @@ func (r *ComponentDependencyUpdateReconciler) removePipelineFinalizer(ctx contex
+		return "", "", err
+	}
+	log.Info("secrets linked to service account", "count", len(linkedSecretNames))
+
+	// get all docker config json secrets
+	allImageRepoSecrets := &corev1.SecretList{}
+	opts := client.ListOption(&client.MatchingFields{"type": string(corev1.SecretTypeDockerConfigJson)})
+
+	if err := r.Client.List(ctx, allImageRepoSecrets, client.InNamespace(namespace), opts); err != nil {
+		return "", "", fmt.Errorf("failed to list secrets of type %s in %s namespace: %w", corev1.SecretTypeDockerConfigJson, namespace, err)
+	}
+	log.Info("found docker config secrets secrets", "count", len(allImageRepoSecrets.Items))
+
+	type DockerConfigJson struct {
+		ConfigAuths map[string]RepositoryConfigAuth `json:"auths"`
+	}
+
+	filteredSecretsData := []RepositoryCredentials{}
+	for _, secret := range allImageRepoSecrets.Items {
+		isSecretLinked := false
+
+		for _, linkedSecret := range linkedSecretNames {
+			if secret.Name == linkedSecret {
+				isSecretLinked = true
+				break
+			}
+		}
+		if !isSecretLinked {
+			continue
+		}
+
+		dockerConfigObject := &DockerConfigJson{}
+		if err = json.Unmarshal(secret.Data[corev1.DockerConfigJsonKey], dockerConfigObject); err != nil {
+			log.Error(err, fmt.Sprintf("unable to parse docker json config in the secret %s", secret.Name))
+			continue
+		}
+
+		for repoName, repoAuth := range dockerConfigObject.ConfigAuths {
+			if repoAuth.Username != "" && repoAuth.Password != "" {
+				filteredSecretsData = append(filteredSecretsData, RepositoryCredentials{SecretName: secret.Name, RepoName: repoName, UserName: repoAuth.Username, Password: repoAuth.Password})
+			} else {
+				if repoAuth.Auth == "" {
+					log.Error(fmt.Errorf("password and username and auth are empty in auth config for repository %s", repoName), "no valid auth")
+					continue
+				} else {
+					decodedAuth, err := base64.StdEncoding.DecodeString(repoAuth.Auth)
+					if err != nil {
+						log.Error(err, fmt.Sprintf("unable to decode docker config json auth for repository %s in the secret %s", repoName, secret.Name))
+						continue
+					}
+					authParts := strings.Split(string(decodedAuth), ":")
+					filteredSecretsData = append(filteredSecretsData, RepositoryCredentials{SecretName: secret.Name, RepoName: repoName, UserName: authParts[0], Password: authParts[1]})
+				}
+			}
+		}
+	}
+
+	imageRepositoryUsername, imageRepositoryPassword, err := r.GetCredentialsForImageRepository(ctx, updatedOutputImage, filteredSecretsData)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("unable to find credential for repository %s", updatedOutputImage))
+		return "", "", err
+	}
+	return imageRepositoryUsername, imageRepositoryPassword, nil
+}
+
+// GetCredentialsForImageRepository This method returns credentials for image repository
+func (r ComponentDependencyUpdateReconciler) GetCredentialsForImageRepository(ctx context.Context, outputImage string, imageRepoSecrets []RepositoryCredentials) (string, string, error) {
```

Comment:
Seems that it doesn't need to be a method

Reply:
sure